scalapact-sbtplugin/src/main/scala/com/itv/scalapact/plugin/ScalaPactPlugin.scala

```diff
@@ -9,30 +9,32 @@ import sbt._
 
 import scala.language.implicitConversions
 
-object ScalaPactPlugin extends Plugin {
+object ScalaPactPlugin extends AutoPlugin {
+  override def requires = plugins.JvmPlugin
+  override def trigger = allRequirements
 
-  val providerStateMatcher = SettingKey[PartialFunction[String, Boolean]]("provider-state-matcher", "Alternative partial function for provider state setup")
-  val providerStates = SettingKey[Seq[(String, String => Boolean)]]("provider-states", "A list of provider state setup functions")
-  val pactBrokerAddress = SettingKey[String]("pactBrokerAddress", "The base url to publish / pull pact contract files to and from.")
-  val providerName = SettingKey[String]("providerName", "The name of the service to verify")
-  val consumerNames = SettingKey[Seq[String]]("consumerNames", "The names of the services that consume the service to verify")
-  val versionedConsumerNames = SettingKey[Seq[(String, String)]]("versionedConsumerNames", "The name and pact version numbers of the services that consume the service to verify")
-  val pactContractVersion = SettingKey[String]("pactContractVersion", "The version number the pact contract will be published under. If missing or empty, the project version will be used.")
-  val allowSnapshotPublish = SettingKey[Boolean]("allowSnapshotPublish", "Flag to permit publishing of snapshot pact files to pact broker. Default is false.")
+  object autoImport {
+    val providerStateMatcher = SettingKey[PartialFunction[String, Boolean]]("provider-state-matcher", "Alternative partial function for provider state setup")
+    val providerStates = SettingKey[Seq[(String, String => Boolean)]]("provider-states", "A list of provider state setup functions")
+    val pactBrokerAddress = SettingKey[String]("pactBrokerAddress", "The base url to publish / pull pact contract files to and from.")
+    val providerName = SettingKey[String]("providerName", "The name of the service to verify")
+    val consumerNames = SettingKey[Seq[String]]("consumerNames", "The names of the services that consume the service to verify")
+    val versionedConsumerNames = SettingKey[Seq[(String, String)]]("versionedConsumerNames", "The name and pact version numbers of the services that consume the service to verify")
+    val pactContractVersion = SettingKey[String]("pactContractVersion", "The version number the pact contract will be published under. If missing or empty, the project version will be used.")
+    val allowSnapshotPublish = SettingKey[Boolean]("allowSnapshotPublish", "Flag to permit publishing of snapshot pact files to pact broker. Default is false.")
+  }
 
+  import autoImport._
 
   private val pactSettings = Seq(
     providerStateMatcher := PartialFunction { (_: String) => false },
     providerStates := Seq(),
-    pactBrokerAddress := "",
-    providerName := "",
     consumerNames := Seq.empty[String],
     versionedConsumerNames := Seq.empty[(String,String)],
-    pactContractVersion := "",
```

Comment:
Setting the defaults to blank always bothered me so this looks like a good idea, but have you checked if the rest of the code does anything like `if(pactBrokerAddress.isEmpty) ...`?